go/worker/event/server.go

```diff
@@ -352,28 +352,28 @@ func Main() error {
 	go func() {
 		slog.Info("worker listening on UNIX domain socket", "socket", sockPath)
 		err := udsServer.Serve(ln)
-		if err != nil && err != http.ErrServerClosed {
-			errorChannel <- fmt.Errorf("UNIX domain socket server failed: %w", err)
-		}
 		// Serve() always returns a non-nil error, so this should not be reachable
 		if err == nil {
 			slog.Error("Serve returned nil", "server", "uds")
 			panic(err)
 		}
+		if err != nil && err != http.ErrServerClosed {
```

Comment:
If did didn't panic above, don't we know err is != nil already?  No need to check again, right?